noah/scripts/articleView.js

```diff
@@ -0,0 +1,133 @@
+};
+
+articleView.handleAuthorFilter = () => {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-author="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = () => {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = () => {
+  $('.main-nav').on('click', '.tab', function(e) {
+    e.preventDefault();
+    $('.tab-content').hide();
+    $(`#${$(this).data('content')}`).fadeIn();
+  });
+
+  $('.main-nav .tab:first').click();
+};
+
+articleView.setTeasers = () => {
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('article').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    if ($(this).text() === 'Read on →') {
+      $(this).parent().find('*').fadeIn();
+      $(this).html('Show Less &larr;');
+    } else {
+      $('body').animate({
+        scrollTop: ($(this).parent().offset().top)
+      },200);
+      $(this).html('Read on &rarr;');
+      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+    }
+  });
+};
+
+// COMMENT: Where is this function called? Why?
+// This function is called within the HTML instead of being included within a javascript file. This is because this function alone needs to apply to more than one page.
+articleView.initNewArticlePage = () => {
+  // TODO (COMPLETED): Ensure the main .tab-content area is revealed. We might add more tabs later or otherwise edit the tab navigation.
+  $('.tab-content').show();
+
+  // TODO (COMPLETED): The new articles we create will be copy/pasted into our source data file.
+  // Set up this "export" functionality. We can hide it for now, and show it once we have data to export.
+
+  $('#article-json').on('focus', function(){
+    $(this).select();
+    $document.execCommand('copy');
+
+  });
+
+  // TODO (COMPLETED): Add an event handler to update the preview and the export field if any inputs change.
+  $('#new-article').on('change', 'input', 'textarea', articleView.create);
+
+};
+
+articleView.create = () => {
+  // TODO (COMPLETED): Set up a variable to hold the new article we are creating.
+  // Clear out the #articles element, so we can put in the updated preview
+    $('#articles > *').remove(); //this gets all the article's children and removes them
+
+
+  // TODO (COMPLETED): Instantiate an article based on what's in the form fields:
+  let articleObj = {
+    title : $('#title').val(),
+    category : $('#category').val(),
+    author : $('#author').val(),
+    authorUrl : $('#authorUrl').val(),
+    publishedOn : $('#publishedOn').val(),
```

Comment:
well need to target the checkbox differently than the other elements
$('#publishedOn:checked').length() will return a truthy or falsey value we can use to assign a value or not